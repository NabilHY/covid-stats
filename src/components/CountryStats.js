import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { MdLocalHospital } from 'react-icons/md';
import { BiCurrentLocation } from 'react-icons/bi';
import { TbRadioactive } from 'react-icons/tb';
import { GiConfirmed } from 'react-icons/gi';
import { FaSkullCrossbones } from 'react-icons/fa';

import axios from 'axios';
import { selectCountry, removeSelectedCountry } from '../redux/actions/countryActions';

const CountryStats = () => {
  const country = useSelector((state) => state.country);
  const { location } = useParams();
  const dispatch = useDispatch();
  const fetchCountryStats = async () => {
    const response = await axios
      .get(`https://covid2019-api.herokuapp.com/v2/country/${location}`)
      .catch(() => { });
    dispatch(selectCountry(response.data.data));
  };

  let loaded = false;

  const change = () => {
    loaded = true;
  };

  useEffect(() => {
    if (location && location !== '' && !loaded) {
      fetchCountryStats();
      change();
    }
    return () => {
      dispatch(removeSelectedCountry());
    };
  }, [location]);

  return (
    <>
      {country[0] && country.map((coun) => (
        <>
          <div key={coun.location} className="metrics">
            <BiCurrentLocation size={30} />
            <h1>{coun.location}</h1>
            <div>
              <div className="info">
                <FaSkullCrossbones size={30} />
                <p>{coun.deaths}</p>
              </div>
              <div className="info">
                <GiConfirmed size={30} />
                <p>{coun.confirmed}</p>
              </div>
              <div className="info">
                <TbRadioactive size={30} />
                <p>{coun.active}</p>
              </div>
              <div className="info">
                <MdLocalHospital size={30} />
                <p>{coun.recovered}</p>
              </div>
            </div>
          </div>
        </>
      ))}
    </>
  );
};

export default CountryStats;
