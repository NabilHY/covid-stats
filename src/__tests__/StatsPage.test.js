import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import CountryStats from '../components/CountryStats';

describe('metrics data rendering', () => {
  test('Country stats page renders', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <CountryStats />
          </Provider>
        </Router>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
