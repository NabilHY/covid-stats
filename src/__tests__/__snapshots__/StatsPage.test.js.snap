// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metrics data rendering Country stats page renders 1`] = `null`;
